src/PostgREST/Types.hs

```diff
@@ -134,8 +133,66 @@ data Proxy = Proxy {
 , proxyPath       :: Text
 } deriving (Show, Eq)
 
-type Operator = Text
-data FValue = VText Text | VForeignKey QualifiedIdentifier ForeignKey deriving (Show, Eq)
+data Operator = Equals | Gte | Gt | Lte | Lt | Neq | Like | ILike | Is | IsNot |
+                TSearch | Contains | Contained | In | NotIn deriving (Eq, Enum)
+
+instance Show Operator where
+  show op =  case op of
+    Equals -> "eq"
+    Gte -> "gte"
+    Gt -> "gt"
+    Lte -> "lte"
+    Lt -> "lt"
+    Neq -> "neq"
+    Like -> "like"
+    ILike -> "ilike"
+    In -> "in"
+    NotIn -> "notin"
+    IsNot -> "isnot"
+    Is -> "is"
+    TSearch -> "@@"
+    Contains -> "@>"
+    Contained -> "<@"
+
+instance Read Operator where
+  readsPrec _ op =  case op of
+    "eq" -> [(Equals, "")]
+    "gte" -> [(Gte, "")]
+    "gt" -> [(Gt, "")]
+    "lte" -> [(Lte, "")]
+    "lt" -> [(Lt, "")]
+    "neq" -> [(Neq, "")]
+    "like" -> [(Like, "")]
+    "ilike" -> [(ILike, "")]
+    "in" -> [(In, "")]
+    "notin" -> [(NotIn, "")]
+    "isnot" -> [(IsNot, "")]
+    "is" -> [(Is, "")]
+    "@@" -> [(TSearch, "")]
+    "@>" -> [(Contains, "")]
+    "<@" -> [(Contained, "")]
+    _ -> []
```

Comment:
This doesn't respect the readsPrec method implementation but I thought of leaving it that way since I only use ```read``` for single values, previously I had:

```haskell
instance Read Operator where
  readsPrec _ op =  case op of
    'e':'q':r -> [(Equals, r)]
    'g':'t':'e':r -> [(Gte, r)]
    'g':'t':r -> [(Gt, r)]
    'l':'t':'e':r -> [(Lte, r)]
    'l':'t':r -> [(Lt, r)]
    'n':'e':'q':r -> [(Neq, r)]
    'l':'i':'k':'e':r -> [(Like, r)]
    'i':'l':'i':'k':'e':r -> [(ILike, r)]
    'i':'n':r -> [(In, r)]
    'n':'o':'t':'i':'n':r -> [(NotIn, r)]
    'i':'s':'n':'o':'t':r -> [(IsNot, r)]
    'i':'s':r -> [(Is, r)]
    '@':'@':r -> [(TSearch, r)]
    '@':'>':r -> [(Contains, r)]
    '<':'@':r -> [(Contained, r)]
    _ -> []
```

I couldn't come up with a way of making this more readable.